### PR TITLE
Fix a crash when accessing a corrupted classic file.

### PR DESCRIPTION
**Pull request by DennisHeimbigner**
* Fixes https://github.com/Unidata/netcdf-c/issues/2731

A corrupted classic file is causing the library to crash. Fix so that it returns NC_ENOTNC instead.

### CODE REVIEW

File: libsrc/posixio.c [hunk 1]
```diff
@@ -536,9 +536,8 @@ px_get(ncio *const nciop, ncio_px *const pxp,
 	off_t diff = (size_t)(offset - blkoffset);
 	off_t blkextent = _RNDUP(diff + extent, pxp->blksz);
 
-	assert(extent != 0);
-	assert(extent < X_INT_MAX); /* sanity check */
-	assert(offset >= 0); /* sanity check */
+	if(extent == 0 || extent >= X_INT_MAX || offset >= 0) /* sanity check */
```
Comment (t-b) [inline libsrc/posixio.c:539]: Drive-by-review:

It should be `offset < 0` right?

Comment (DennisHeimbigner) [reply libsrc/posixio.c]: Correct; just pushed a fix.

**Review by WardF — APPROVED** [review]
